packages/protocol/test/verifiers/RiscZeroVerifier.t.sol

```diff
@@ -0,0 +1,187 @@
+        vm.stopPrank();
+
+        // Caller not necessary has to be TaikoL1 contract because there is no keys (as in SGX keys)
+        // to be front run.
+        vm.startPrank(Alice);
+
+        // Context
+        IVerifier.Context memory ctx = IVerifier.Context({
+            metaHash: bytes32("ab"),
+            blobHash: bytes32("cd"),
+            prover: address(0),
+            msgSender: address(0),
+            blockId: 10,
+            isContesting: false,
+            blobUsed: false
+        });
+
+        // Transition
+        TaikoData.Transition memory transition = TaikoData.Transition({
+            parentHash: bytes32("12"),
+            blockHash: bytes32("34"),
+            stateRoot: bytes32("56"),
+            graffiti: bytes32("78")
+        });
+
+        bytes memory seal = hex"00";
+        bytes32 imageId = bytes32("11");
+        bytes32 postStateDigest = bytes32("22");
+
+        // TierProof
+        TaikoData.TierProof memory proof =
+            TaikoData.TierProof({ tier: 100, data: abi.encode(seal, imageId, postStateDigest) });
+
+        vm.warp(block.timestamp + 5);
+
+        // `verifyProof()`
+        rv.verifyProof(ctx, transition, proof);
+
+        vm.stopPrank();
+    }
+
+    function test_verifyProof_invalidImageId() external {
+        vm.stopPrank();
+
+        // Caller not necessary has to be TaikoL1 contract because there is no keys (as in SGX keys)
+        // to be front run.
+        vm.startPrank(Alice);
+
+        // Context
+        IVerifier.Context memory ctx = IVerifier.Context({
+            metaHash: bytes32("ab"),
+            blobHash: bytes32("cd"),
+            prover: address(0),
+            msgSender: address(0),
+            blockId: 10,
+            isContesting: false,
+            blobUsed: false
+        });
+
+        // Transition
+        TaikoData.Transition memory transition = TaikoData.Transition({
+            parentHash: bytes32("12"),
+            blockHash: bytes32("34"),
+            stateRoot: bytes32("56"),
+            graffiti: bytes32("78")
+        });
+
+        bytes memory seal = hex"00";
+        bytes32 imageId = bytes32("121");
+        bytes32 postStateDigest = bytes32("22");
+
+        // TierProof
+        TaikoData.TierProof memory proof =
+            TaikoData.TierProof({ tier: 100, data: abi.encode(seal, imageId, postStateDigest) });
+
+        vm.warp(block.timestamp + 5);
+
+        vm.expectRevert(RiscZeroVerifier.RISC_ZERO_INVALID_IMAGE_ID.selector);
+        rv.verifyProof(ctx, transition, proof);
+
+        vm.stopPrank();
+    }
+
+    function test_verifyProof_invalidProof() external {
+        riscZeroRemoteVerifier.setVerifier(false);
+        vm.stopPrank();
+
+        // Caller not necessary has to be TaikoL1 contract because there is no keys (as in SGX keys)
+        // to be front run.
+        vm.startPrank(Alice);
+
+        // Context
+        IVerifier.Context memory ctx = IVerifier.Context({
```

Comment:
most of the code in these tests are the setup. We can make the overall tests simplier if we merge these test togeter by using the same set up code, or define functions like "getRandomContext / getRandmTransition".